syntax = "proto3";

message SearchRequest {
    string query = 1;
    int32 page_number = 2;
    int32 result_per_page = 3;
}

message Result {
    string url = 1;
    string title = 2;
    repeated string snippets = 3;
}

message SearchResponse {
    repeated Result results = 1;
}

message HelloRequest{
    string name = 1;
}

message HelloReply{
    string msg = 1;
}

service SearchService {
  rpc Search (SearchRequest) returns (SearchResponse);
  rpc SayHello (HelloRequest) returns (HelloReply);
}